---
title: "tracking_predicted_ma_virus_vps"
author: "A.J Wing"
date: "2023-02-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

## import metadata
```{r}
metadata <- read_tsv("~/Desktop/erie_story_r_work/2014_story_metadata_updated_8_9_22.tsv")
metadata$samples <- gsub("Sample_","",metadata$samples)
```

```{r}
both_peaks_ma_virus_tpm_norm_counts <- read_tsv("~/Desktop/erie_story_r_work/both_peaks_ma_virus_tpm_norm_counts.tsv", col_names=T)
```



```{r}
clusters <- read_tsv("~/Desktop/erie_story_r_work/merged_3kb_trimmed_viruses_and_ma_virus_references_95-85_clusters_parsed.tsv", col_names=T)
```
```{r}
clusters_3000 <- clusters %>%
  filter(length >= 3000)
```

##only use this for adding vp info to all predicted ma phages (not for beta diversity)
```{r}
names(clusters)[names(clusters) == 'contig'] <- "phage"
```
##only use this for binding vp info to all predicted ma phages (not for beta diversity)
```{r}
microcystis_viruses_with_vp_and_counts <- left_join(both_peaks_ma_virus_tpm_norm_counts,clusters,by="phage")
dim(microcystis_viruses_with_vp_and_counts)
```
##only use this for cleaning microcystis viruses with vps (not for beta diversity)
```{r}
ma_viruses_with_vp_and_tpm <- microcystis_viruses_with_vp_and_counts[, c("cluster", "phage", "TPM", "Length", "samples", "Predictions", "InfProbabilities")]
```
##only use this for cleaning microcystis viruses with vps (not for beta diversity)
```{r}
ma_viruses_with_vp_and_tpm <- ma_viruses_with_vp_and_tpm[!duplicated(ma_viruses_with_vp_and_tpm$phage), ]

ma_viruses_with_vp_and_tpm <- ma_viruses_with_vp_and_tpm[ma_viruses_with_vp_and_tpm$Predictions==1,]

#ma_viruses_with_vp_and_tpm <- ma_viruses_with_vp_and_tpm[ma_viruses_with_vp_and_tpm$Length>=10000,]

count_ma_virus_vps <- ma_viruses_with_vp_and_tpm %>%
                              count(cluster)

```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}
metadata_for_ma_virus_vps <- metadata
```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}

ma_viruses_with_vp_and_tpm$samples <-as.character(ma_viruses_with_vp_and_tpm$samples)
metadata_for_ma_virus_vps$samples <-as.character(metadata_for_ma_virus_vps$samples)


ma_viruses_with_vp_and_tpm <- left_join(ma_viruses_with_vp_and_tpm,metadata_for_ma_virus_vps,by="samples")
dim(ma_viruses_with_vp_and_tpm)
```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}
VP_number_dates <- read_tsv("~/Desktop/erie_story_r_work/VP_number_of_sample_dates.tsv", col_names=T)
colnames(VP_number_dates)[1] <- "cluster"
```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}
ma_viruses_with_vp_and_tpm <- left_join(ma_viruses_with_vp_and_tpm,VP_number_dates,by="cluster")
dim(ma_viruses_with_vp_and_tpm)
```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}
ma_viruses_with_vp_tpm_num_dates <- ma_viruses_with_vp_and_tpm[, c("cluster", "phage", "TPM", "number_dates", "Length", "Date", "Station", "fraction", "samples", "Predictions", "InfProbabilities")]
```
##only use this for microcystis vp tracking (not for beta diversity)
```{r}
ma_viruses_with_vp_tpm_num_dates <- ma_viruses_with_vp_tpm_num_dates[ma_viruses_with_vp_tpm_num_dates$Length>=10000,]
```



#remove nas
```{r}
ma_viruses_with_vp_tpm_num_dates <- na.omit(ma_viruses_with_vp_tpm_num_dates)
```

#count number of unique vps in dataset
```{r}
length(unique(ma_viruses_with_vp_tpm_num_dates$cluster))
```

#count number of vps for each station in total
```{r}
table(ma_viruses_with_vp_tpm_num_dates$Station)
```

#distribution of vps and the number of dates they were present in
```{r}
table(ma_viruses_with_vp_tpm_num_dates$number_dates)
```

##try collapsing clusters and keeping max length of each
```{r}
rep_ma_viruses_full <- ma_viruses_with_vp_tpm_num_dates %>% group_by(cluster) %>% slice(which.max(Length))
```

```{r}
write_tsv(rep_ma_viruses_full, "~/Desktop/erie_story_r_work/predicted_ma_virus_vps.tsv", col_names = TRUE)
```

##import taxonomy
```{r}
custom_taxonomy <- read_tsv("~/Desktop/erie_story_r_work/all_merged_3000_trimmed_viruses_plus_known_ma_virus_seqs_frags_and_sep_lem_variant.PTT.virus-taxonomy.tsv")
names(custom_taxonomy)[names(custom_taxonomy) == 'scaffold'] <- "phage"
```

#now add in taxonomy
```{r}
rep_ma_viruses_full_w_taxonomy <- left_join(rep_ma_viruses_full,custom_taxonomy,by="phage")
dim(rep_ma_viruses_full_w_taxonomy)
```


## import viral predictions for each bloom


#august bloom predictions without viral binning
```{r}
# load data
HABs2014 <- read_tsv("~/Desktop/erie_story_r_work/Aug4_predictions.tsv")
HABs2014_temp <- separate(HABs2014, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
HABs2014_aug_bloom <- HABs2014_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
HABs2014_aug_bloom$phage <- gsub(".fasta", "", HABs2014_aug_bloom$phage) 
HABs2014_aug_bloom <-HABs2014_aug_bloom[!(HABs2014_aug_bloom$host=="Sample_49613_42.fasta"),]
HABs2014_aug_bloom <-HABs2014_aug_bloom[!(HABs2014_aug_bloom$host=="Sample_49613_9.fasta"),]
```

#september bloom
```{r}
# load data
HABs2014 <- read_tsv("~/Desktop/erie_story_r_work/Sep29_predictions.tsv")
HABs2014_temp <- separate(HABs2014, col=pairs, c("phage", "host"), sep=":concatenated_", remove = FALSE)
HABs2014_sep_bloom <- HABs2014_temp[, c("phage", "host", "Homology", "Predictions", "GCdiff", "k3dist", "k6dist", "InfProbabilities")]
HABs2014_sep_bloom$phage <- gsub(".fasta", "", HABs2014_sep_bloom$phage) 
#HABs2014_sep_bloom$phage <- gsub("concatenated_", "", HABs2014$phage)
head(HABs2014_sep_bloom)
```

##now bind aug4 and sep29 ma virus tpm norm'd counts
```{r}
both_peaks_predictions <- rbind(HABs2014_aug_bloom,HABs2014_sep_bloom)
dim(both_peaks_predictions)
```

#now apply threshold to reduce network size while increasing confidence in predictions
```{r}
thresh <- 0.93

both_peaks_predictions <- both_peaks_predictions[which(both_peaks_predictions$InfProbabilities > thresh), ]
print(dim(both_peaks_predictions))
```
##get number of hosts per viral population
```{r}
uniq_phage <- unique(both_peaks_predictions$phage)
number_host <- vector(mode = "integer", length(uniq_phage))

for(i in 1:length(uniq_phage)){
  cur <- uniq_phage[i]
  temp <- both_peaks_predictions[which(both_peaks_predictions$phage == cur), ]
  number_host[i] <- nrow(temp)
}

number_hosts <- data.frame(uniq_phage, number_host)
number_hosts
```

#combine number of hosts with predicted ma viruses
```{r}
names(number_hosts)[names(number_hosts) == 'uniq_phage'] <- "phage"

rep_ma_viruses_full_w_taxonomy_num_hosts <- left_join(rep_ma_viruses_full_w_taxonomy,number_hosts,by="phage")
dim(rep_ma_viruses_full_w_taxonomy_num_hosts)
```

```{r}
ggplot(rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=number_host)) +
  geom_histogram(binwidth = 1, color = 'white')

ggplot(data=rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=number_host)) +
    geom_bar(aes(y= (..count..)/sum(..count..)), fill = '#20A387FF') +
    theme(panel.background = element_blank(),
          axis.title=element_text(size=20)) +
    guides(size=FALSE) + xlab("Number of Hosts") + ylab("% of Viruses Infecting Hosts")

ggplot(data=rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=number_host)) +
    geom_bar(fill = '#20A387FF') +
    theme(panel.background = element_blank(),
          axis.title=element_text(size=20)) +
    guides(size=FALSE) + xlab("Number of Hosts") + ylab("Number of Viruses Infecting Hosts")
```

###plot contig length distribution
```{r}
length_distribution <- ggplot(rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=Length)) +
  geom_histogram(binwidth = 1000, fill = '#287D8EFF') +
  xlim(10000,255000) +
  #scale_fill_jcolors(palette = "default") +
  ggtitle("Microcystis virus populations (min. 10kb)") +
  xlab("Genome Length (binwidth=1000)") +
  ylab("Number of Populations") +
  guides(colour = guide_legend(override.aes = list(size=10))) +
  theme(axis.text.x = element_text(angle =45, hjust = 1)) +
  theme(panel.background = element_blank(),
        axis.line = element_line(colour = "black"), 
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        strip.text = element_text(size = 18),
        strip.background = element_rect(colour= "black", fill = "lemon chiffon", linetype= "solid"),
        axis.title=element_text(size=18),
        legend.title = element_text(size = 16),
        legend.text = element_text(size = 14),
        plot.title = element_text(size = 24, hjust = 0.5, colour = "black"),
        axis.text.y = element_text(size=14, colour = "black"),
        axis.text.x  = element_text(size=14, colour="black"),
        plot.margin = unit(c(1, 1, 1, 1),"lines"))
        plot.margin = margin(t = 1,  # Top margin
                             r = 1,  # Right margin
                             b = 3,  # Bottom margin
                             l = 2,  # Left margin
                             unit = "cm")

length_distribution

```

```{r}
family_plot <- ggplot(data=rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=family, color=family)) +
    geom_bar(aes(y= (..count..), fill=family)) +
    #coord_flip() +
  theme(panel.background = element_blank(), 
        axis.line = element_line(colour = "black"), 
        panel.border = element_rect(colour = "black", fill=NA, size=1), 
        axis.title=element_text(size=16), 
        axis.text.y = element_text(size=12, colour = "black"),
        axis.text.x  = element_text(size=12, angle=45, vjust=1, hjust=1, colour="black"),
        plot.margin = unit(c(0.5, 1, 5, 0.6),"lines"),
        #legend.position = "bottom", 
        legend.direction = "horizontal",
        legend.text = element_text(size=8),
        legend.position = c(0.2,-1.7)) +
  guides(fill=guide_legend(nrow=3, byrow=T), color=guide_legend(nrow=3, byrow=T)) +
  scale_colour_manual(values = c('Ackermannviridae' = 'yellow', 'ambiguous' = 'gray', 'Herelleviridae' = '#482677FF', 'Microviridae' = 'red', 'Myoviridae' = '#55C667FF', 'Podoviridae' = '#287D8EFF', 'Siphoviridae' = '#DCE319FF', 'unknown' = 'gray', 'unassigned' = 'gray' )) +
  scale_fill_manual(name = '',
                     values = c('Ackermannviridae' = 'yellow', 'ambiguous' = 'gray', 'Herelleviridae' = '#482677FF', 'Microviridae' = 'red', 'Myoviridae' = '#55C667FF', 'Podoviridae' = '#287D8EFF', 'Siphoviridae' = '#DCE319FF', 'unknown' = 'gray', 'unassigned' = 'gray' )) +
  xlab("Predicted Microcystis Virus Families") +
  ylab("Count") 

family_plot
```
```{r}
# Reorder the factor levels for Date so that 4-Aug-14 comes before 29-Sep-14
rep_ma_viruses_full_w_taxonomy_num_hosts$fraction <- factor(rep_ma_viruses_full_w_taxonomy_num_hosts$fraction, 
                                                        levels = c("Viral", "Whole", "3", "53", "100"))

family_plot <- ggplot(data = rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x = family)) +
  geom_bar(aes(y = ..count..), fill = "#287D8EFF", color = "#287D8EFF") +
  facet_wrap(~fraction) +
  theme(
    panel.background = element_blank(), 
    axis.line = element_line(colour = "black"), 
    panel.border = element_rect(colour = "black", fill = NA, size = 1), 
    axis.title = element_text(size = 16), 
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.text.x = element_text(size = 12, angle = 90, vjust = 1, hjust = 1, colour = "black"),
    plot.margin = unit(c(0.5, 1, 5, 0.6), "lines"),
    legend.position = "none"
  ) +
  xlab("Predicted Microcystis Virus Families") +
  ylab("Count")

family_plot

```


```{r}
# Reorder the factor levels for Date so that 4-Aug-14 comes before 29-Sep-14
rep_ma_viruses_full_w_taxonomy_num_hosts$Date <- factor(rep_ma_viruses_full_w_taxonomy_num_hosts$Date, 
                                                        levels = c("4-Aug-14", "29-Sep-14"))

family_plot <- ggplot(data = rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x = family)) +
  geom_bar(aes(y = ..count..), fill = "#287D8EFF", color = "#287D8EFF") +
  facet_wrap(~Date) +
  theme(
    panel.background = element_blank(), 
    axis.line = element_line(colour = "black"), 
    panel.border = element_rect(colour = "black", fill = NA, size = 1), 
    axis.title = element_text(size = 16), 
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.text.x = element_text(size = 12, angle = 45, vjust = 1, hjust = 1, colour = "black"),
    plot.margin = unit(c(0.5, 1, 5, 0.6), "lines"),
    legend.position = "none"
  ) +
  xlab("Predicted Microcystis Virus Families") +
  ylab("Count")

family_plot

```


```{r}
date_plot <- ggplot(data=rep_ma_viruses_full_w_taxonomy_num_hosts, aes(x=number_dates)) +
    geom_bar(fill = '#287D8EFF') +
    theme(panel.background = element_blank(),
          axis.title=element_text(size=20)) +
    guides(size=FALSE) + xlab("Number of Dates Microcystis Virus Population In") + ylab("Number of Populations") 

date_plot
```

##make taxonomy database plot to show any biases in our own taxonomy plot
```{r}
db_taxonomy <- read_tsv("~/Desktop/erie_story_r_work/PTT_virus_taxonomy.tsv")

```

```{r}
db_family_plot <- ggplot(data=db_taxonomy, aes(x=Family)) +
  geom_bar(aes(y = ..count..), fill = "#287D8EFF", color = "#287D8EFF") +
    #coord_flip() +
  theme(panel.background = element_blank(), 
        axis.line = element_line(colour = "black"), 
        panel.border = element_rect(colour = "black", fill=NA, size=1), 
        axis.title=element_text(size=16), 
        axis.text.y = element_text(size=12, colour = "black"),
        axis.text.x  = element_text(size=12, angle=45, vjust=1, hjust=1, colour="black"),
        plot.margin = unit(c(0.5, 1, 5, 0.6),"lines"),
        #legend.position = "bottom", 
        legend.direction = "horizontal",
        legend.text = element_text(size=8),
        legend.position = c(0.2,-1.7)) +
  xlab("Predicted Microcystis Virus Families") +
  ylab("Count") 

db_family_plot
```



##now lets do some orf exploration with our predicted microcystis viruses between peaks
## import DRAM annotations

```{r}
annotations <- read_tsv("~/Desktop/erie_story_r_work/merged_annotations.tsv", col_names = T)
colnames(annotations)[2] <- "Geneid"
colnames(annotations)[4] <- "phage"
colnames(annotations)[5] <- "Start"
colnames(annotations)[7] <- "End"
annotations <- annotations[,-1]
annotations <- separate(annotations, col="phage", into=c("Assembly", "Contig"), sep = "--", remove = F)

```


##combine orf annotations with peak ma viruses
```{r}
rep_ma_viruses_orfs <- left_join(rep_ma_viruses_full_w_taxonomy_num_hosts, annotations, by="phage")
dim(rep_ma_viruses_orfs)
```

##lets simplify our data table so we remove some columns that aren't pertinent to our goal of plotting pfam annotations
```{r}
rep_ma_viruses_orfs_cleaned <- rep_ma_viruses_orfs[, c("Date", "kegg_id", "pfam_hits", "vogdb_categories")]
```

```{r}
library(data.table)

peak_info_kegg_metabolism <- readLines("~/Desktop/erie_story_r_work/kegg_metabolism.txt")
peak_info_kegg_metabolism <- str_split(peak_info_kegg_metabolism, pattern = "\t", n = 4)
peak_info_kegg_metabolism <- data.frame(peak_info_kegg_metabolism)
peak_info_kegg_metabolism <- data.frame(t(peak_info_kegg_metabolism))
rownames(peak_info_kegg_metabolism) <- NULL
temp <- data.frame(tstrsplit(peak_info_kegg_metabolism[,4], split = "\t"))
 peak_info_kegg_metabolism <- cbind.data.frame(peak_info_kegg_metabolism[,1:3], temp)
colnames(peak_info_kegg_metabolism) <- c("Level1", "Level2", "Level3", "kegg_id", "Level4")
```

```{r}
rep_ma_viruses_orfs_pfam_kegg <- left_join(rep_ma_viruses_orfs_cleaned, peak_info_kegg_metabolism, by="kegg_id")
dim(rep_ma_viruses_orfs_pfam_kegg)
```


```{r}
rep_ma_viruses_orfs_pfam_kegg <- rep_ma_viruses_orfs_pfam_kegg %>%
  group_by(pfam_hits) %>%
  summarize(n = n(), 
            prop_4aug = sum(Date == "4-Aug-14") / n(), 
            prop_29sep = sum(Date == "29-Sep-14") / n(),
            dates_found = case_when(
              sum(Date == "4-Aug-14") > 0 & sum(Date == "29-Sep-14") == 0 ~ "4-Aug-14",
              sum(Date == "29-Sep-14") > 0 & sum(Date == "4-Aug-14") == 0 ~ "29-Sep-14",
              sum(Date == "4-Aug-14") > 0 & sum(Date == "29-Sep-14") > 0 ~ "both_dates",
              TRUE ~ "other"
            )) %>%
  left_join(rep_ma_viruses_orfs_pfam_kegg, by = "pfam_hits")

```

```{r}
ma_viruses_orfs_pfam <- rep_ma_viruses_orfs_pfam_kegg[, c("pfam_hits", "vogdb_categories", "dates_found", "Date", "kegg_id", "Level2")]
```

```{r}
# Remove everything after the semi-colon in pfam_hits
ma_viruses_orfs_pfam$pfam_hits <- gsub(";.*", "", ma_viruses_orfs_pfam$pfam_hits)
```

```{r}
# Subset the data frame by date
ma_viruses_orfs_pfam_sep29 <- subset(ma_viruses_orfs_pfam, Date == "29-Sep-14")
ma_viruses_orfs_pfam_aug4 <- subset(ma_viruses_orfs_pfam, Date == "4-Aug-14")

# Print the two data frames
ma_viruses_orfs_pfam_sep29
ma_viruses_orfs_pfam_aug4

```

```{r}
aug4_orf_summary <- tibble(pfam=names(table(ma_viruses_orfs_pfam_aug4$pfam_hits)),
                      aug4_frequency=table(ma_viruses_orfs_pfam_aug4$pfam_hits))

aug4_orf_summary$aug4_proportion_pfam <- aug4_orf_summary$aug4_frequency/(nrow(ma_viruses_orfs_pfam_aug4))

aug4_orf_summary <- aug4_orf_summary[order(aug4_orf_summary$aug4_proportion_pfam, decreasing=T),] 
```

```{r}
sep29_orf_summary <- tibble(pfam=names(table(ma_viruses_orfs_pfam_sep29$pfam_hits)),
                      sep29_frequency=table(ma_viruses_orfs_pfam_sep29$pfam_hits))

sep29_orf_summary$sep29_proportion_pfam <- sep29_orf_summary$sep29_frequency/(nrow(ma_viruses_orfs_pfam_sep29))

sep29_orf_summary <- sep29_orf_summary[order(sep29_orf_summary$sep29_proportion_pfam, decreasing=T),] 
```


```{r}
library(dplyr)

both_peaks_combined_data <- full_join(aug4_orf_summary, sep29_orf_summary, by = "pfam") %>%
  mutate(across(-pfam, ~replace_na(as.numeric(.), 0)))

```

```{r}
kegg_info_peak_orfs <- rep_ma_viruses_orfs_pfam_kegg[, c("pfam_hits", "Level2")]
kegg_info_peak_orfs$pfam_hits <- gsub(";.*", "", kegg_info_peak_orfs$pfam_hits)
kegg_info_peak_orfs <- replace(kegg_info_peak_orfs, kegg_info_peak_orfs == "", NA)
kegg_info_peak_orfs_unique <- kegg_info_peak_orfs %>%
  slice(which(!is.na(Level2))) %>%
  distinct(pfam_hits, .keep_all = TRUE)

```

```{r}
both_peaks_combined_orfs <- left_join(both_peaks_combined_data, kegg_info_peak_orfs_unique, by= c("pfam" = "pfam_hits"))
dim(both_peaks_combined_orfs)
```

```{r}
both_peaks_orf_presence <- both_peaks_combined_orfs %>%
  mutate(dates_found = case_when(aug4_frequency > 0 & sep29_frequency == 0 ~ "aug4",
                                 sep29_frequency > 0 & aug4_frequency == 0 ~ "sep29",
                                 aug4_frequency > 0 & sep29_frequency > 0 ~ "both",
                                 TRUE ~ NA_character_))

```

```{r}
# Specify the file path where you want to save the .tsv file
file_path <- "~/Desktop/erie_story_r_work/both_peaks_annotated_genes_for_first_paper_fig_4_11_8_23.tsv"

# Export the data frame as a .tsv file
write.table(both_peaks_orf_presence, file = file_path, sep = "\t", row.names = FALSE)
```

```{r}
tmp_df <- ma_viruses_orfs_pfam %>%
  select(pfam_hits, vogdb_categories)

vogdb_cats <- both_peaks_orf_presence %>%
  left_join(tmp_df, by = c("pfam" = "pfam_hits"))

vogdb_cats <- vogdb_cats %>%
  distinct()

vogdb_cats <- vogdb_cats %>%
  filter(!is.na(vogdb_categories))

vogdb_cats_xs <- vogdb_cats %>%
  filter(str_detect(vogdb_categories, "Xs"))
```

```{r}
both_peaks_orf_presence <- read_tsv("~/Desktop/erie_story_r_work/both_peaks_annotated_genes_for_first_paper_fig_4_11_8_23.tsv")
```

```{r}

# Replace 0 values with 0.00001
both_peaks_orf_presence$aug4_proportion_pfam[both_peaks_orf_presence$aug4_proportion_pfam == 0] <- 0.00001
both_peaks_orf_presence$sep29_proportion_pfam[both_peaks_orf_presence$sep29_proportion_pfam == 0] <- 0.00001

```


```{r}
# Load required packages
library(ggplot2)

# Create scatterplot
bloom_peak_ma_virus_gene_annotations<- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(size = 4, stroke = 1) +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  scale_x_continuous(limits = c(0.000001, 0.0085), expand = c(0, 0)) +
  scale_y_continuous(limits = c(0.000001, 0.0085), expand = c(0, 0))

bloom_peak_ma_virus_gene_annotations
```

```{r}
png("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_gene_annotations.png", width=8, height=5, units="in", res=300)
bloom_peak_ma_virus_gene_annotations     
dev.off()
```

```{r}
pdf("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_gene_annotations.pdf", width=8, height=5)
bloom_peak_ma_virus_gene_annotations     
dev.off()
```


```{r}
library(ggplot2)
library(ggrepel)

# Create scatterplot
log_bloom_peak_ma_virus_gene_annotations <- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(size = 4, stroke = 1) +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  scale_x_log10(limits = c(0.000005, 0.009), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000005, 0.009), expand = c(0, 0)) 

log_bloom_peak_ma_virus_gene_annotations

```

```{r}
png("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_gene_annotations_log_3_27_23.png", width=8, height=5, units="in", res=300)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```

```{r}
pdf("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_gene_annotations_log_3_27_23.pdf", width=8, height=5)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```


```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_orf_presence$row_num <- seq(nrow(both_peaks_orf_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 8, 9, 16, 23, 24, 32, 52, 59, 539, 540, 541, 544)

log_bloom_peak_ma_virus_gene_annotations <- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(data = subset(both_peaks_orf_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 3, stroke = 1, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_orf_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.05, height = 0.05, alpha = 0.7, size = 3) +
  geom_text_repel(data = both_peaks_orf_presence[both_peaks_orf_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 2, color = "black") +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  scale_x_log10(limits = c(0.000005, 0.009), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000005, 0.009), expand = c(0, 0))

log_bloom_peak_ma_virus_gene_annotations

```

```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_orf_presence$row_num <- seq(nrow(both_peaks_orf_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 8, 9, 16, 23, 24, 32, 52, 59, 539, 540, 541, 544)

log_bloom_peak_ma_virus_gene_annotations <- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(data = subset(both_peaks_orf_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 3, stroke = 1, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_orf_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.07, height = 0.07, alpha = 0.7, size = 3) +
#  geom_text_repel(data = both_peaks_orf_presence[both_peaks_orf_presence$row_num %in% #rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = #as.character(pfam)), size = 3, color = "black") +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  theme(panel.grid = element_blank()) +
  scale_x_log10(limits = c(0.000003, 0.012), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000003, 0.012), expand = c(0, 0))

log_bloom_peak_ma_virus_gene_annotations

```

```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_orf_presence$row_num <- seq(nrow(both_peaks_orf_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 8, 9, 16, 23, 24, 32, 52, 59, 539, 540, 541, 544)

log_bloom_peak_ma_virus_gene_annotations <- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(data = subset(both_peaks_orf_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 3, stroke = 1, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_orf_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.07, height = 0.07, alpha = 0.7, size = 3) +
  geom_text_repel(data = both_peaks_orf_presence[both_peaks_orf_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 3, color = "black") +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  theme(panel.grid = element_blank()) +
  scale_x_log10(limits = c(0.000003, 0.012), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000003, 0.012), expand = c(0, 0))

log_bloom_peak_ma_virus_gene_annotations

```

```{r}
png("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_log_with_labels_12_17_24.png", width=12, height=8, units="in", res=300)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```

```{r}
pdf("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_log_with_labels_12_17_24.pdf", width=12, height=8)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```


```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_orf_presence$row_num <- seq(nrow(both_peaks_orf_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 8, 9, 16, 23, 24, 32, 52, 59, 539, 540, 541, 544)

log_bloom_peak_ma_virus_gene_annotations <- ggplot(data = both_peaks_orf_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = Level2, shape = dates_found, fill = Level2)) +
  geom_point(data = subset(both_peaks_orf_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 3, stroke = 1, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_orf_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.07, height = 0.07, alpha = 0.7, size = 3) +
  geom_label_repel(data = both_peaks_orf_presence[both_peaks_orf_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 3, color = "black") +
  scale_color_discrete(name = "KO Metabolic Module") +
  scale_fill_discrete(name = "KO Metabolic Module") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  scale_x_log10(limits = c(0.000005, 0.009), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000005, 0.009), expand = c(0, 0))

log_bloom_peak_ma_virus_gene_annotations

```

```{r}
png("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_log_color_labels_3_29_23.png", width=8, height=5, units="in", res=300)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```

```{r}
pdf("~/Desktop/erie_story_r_work/bloom_peak_ma_virus_log_color_labels_3_29_23.pdf", width=12, height=8)
log_bloom_peak_ma_virus_gene_annotations     
dev.off()
```


##Try looking only at genes specific to viruses here
```{r}


keywords <- c("gp20", "portal", "gp23", "major capsid", "gp91", "tail sheath", "gp43", "gp101", "tail fibre", "gp51", "base plate", "gp7", "gp10", "gp12", "polymerase family A", "poly A", "hk97", "t4", "t7", "t5", "lambda")
pattern <- paste(keywords, collapse = "|")

# Using subset and grepl:
both_peaks_only_viral_orfs_presence <- subset(both_peaks_orf_presence, grepl(pattern, pfam, ignore.case = TRUE))

```

```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_only_viral_orfs_presence$row_num <- seq(nrow(both_peaks_only_viral_orfs_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 12, 17, 19, 9, 20, 24, 25, 26, 27, 28)

log_bloom_peak_ma_virus_only_gene_annotations <- ggplot(data = both_peaks_only_viral_orfs_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = dates_found, shape = dates_found, fill = dates_found)) +
  geom_point(data = subset(both_peaks_only_viral_orfs_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 3, stroke = 1, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_only_viral_orfs_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.07, height = 0.07, alpha = 0.7, size = 3) +
  geom_text_repel(data = both_peaks_only_viral_orfs_presence[both_peaks_only_viral_orfs_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 3, color = "black") +
  scale_color_discrete(name = "Date Found") +
  scale_fill_discrete(name = "Date Found") +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  theme(panel.grid = element_blank()) +
  scale_x_log10(limits = c(0.000003, 0.012), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000003, 0.012), expand = c(0, 0))

log_bloom_peak_ma_virus_only_gene_annotations

```

```{r}
library(ggplot2)
library(ggrepel)

# Add row numbers to the data frame
both_peaks_only_viral_orfs_presence$row_num <- seq(nrow(both_peaks_only_viral_orfs_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 12, 17, 19, 9, 20, 24, 25, 26, 27, 28)

log_bloom_peak_ma_virus_only_gene_annotations <- ggplot(data = both_peaks_only_viral_orfs_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = dates_found, shape = dates_found, fill = dates_found)) +
  geom_point(data = subset(both_peaks_only_viral_orfs_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 7, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_only_viral_orfs_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.07, height = 0.07, alpha = 0.7, size = 3) +
  #geom_text_repel(data = both_peaks_only_viral_orfs_presence[both_peaks_only_viral_orfs_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 3, color = "black") +
  scale_color_manual(name = "Date Found", values = c("aug4" = "orange", "sep29" = "hotpink", "both" = "gray")) +
  scale_fill_manual(name = "Date Found", values = c("aug4" = "orange", "sep29" = "hotpink", "both" = "gray")) +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  theme(panel.grid = element_blank()) +
  scale_x_log10(limits = c(0.0000000001, 0.012), expand = c(0.05, 0.05)) +
  scale_y_log10(limits = c(0.0000000001, 0.012), expand = c(0.05, 0.05))

log_bloom_peak_ma_virus_only_gene_annotations

```
```{r}
# Add a small pseudocount to handle zeros in the data
pseudocount <- 1e-5 # Small positive value to replace zeros
both_peaks_only_viral_orfs_presence$aug4_proportion_pfam <- 
  ifelse(both_peaks_only_viral_orfs_presence$aug4_proportion_pfam == 0, pseudocount, both_peaks_only_viral_orfs_presence$aug4_proportion_pfam)
both_peaks_only_viral_orfs_presence$sep29_proportion_pfam <- 
  ifelse(both_peaks_only_viral_orfs_presence$sep29_proportion_pfam == 0, pseudocount, both_peaks_only_viral_orfs_presence$sep29_proportion_pfam)


# Add row numbers to the data frame
both_peaks_only_viral_orfs_presence$row_num <- seq(nrow(both_peaks_only_viral_orfs_presence))

# Define the row numbers of the points you want to label
rows_to_label <- c(1, 2, 3, 4, 7, 12, 17, 19, 9, 20, 24, 25, 26, 27, 28)

log_bloom_peak_ma_virus_only_gene_annotations <- ggplot(data = both_peaks_only_viral_orfs_presence, aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, color = dates_found, shape = dates_found, fill = dates_found)) +
  geom_point(data = subset(both_peaks_only_viral_orfs_presence, !is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), size = 5, alpha = 0.9) +
  geom_jitter(data = subset(both_peaks_only_viral_orfs_presence, is.na(Level2)), aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam), width = 0.1, height = 0.1, alpha = 0.7, size = 5) +
  geom_text_repel(data = both_peaks_only_viral_orfs_presence[both_peaks_only_viral_orfs_presence$row_num %in% rows_to_label,], aes(x = aug4_proportion_pfam, y = sep29_proportion_pfam, label = as.character(pfam)), size = 3, color = "black") +
  scale_color_manual(name = "Date Found", values = c("aug4" = "orange", "sep29" = "hotpink", "both" = "gray")) +
  scale_fill_manual(name = "Date Found", values = c("aug4" = "orange", "sep29" = "hotpink", "both" = "gray")) +
  scale_shape_manual(name = "Date Found", values = c("aug4" = 21, "sep29" = 22, "both" = 23)) +
  labs(x = "4 August Pfam Proportion", y = "29 September Pfam Proportion") +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
  theme_bw() + 
  theme(panel.grid = element_blank()) +
  scale_x_log10(limits = c(0.000003, 0.012), expand = c(0, 0)) +
  scale_y_log10(limits = c(0.000003, 0.012), expand = c(0, 0))

log_bloom_peak_ma_virus_only_gene_annotations


```



